### ISSUE TITLE
Fatal Exception: java.lang.OutOfMemoryError

### ISSUE THREAD

**Issue by princeHago** [fabian7593/MagicalCamera#28]
Hi,

i use your MagicalCamera to take several photo to send (encoded 64) to a server.

Sometime i got this error:

> Fatal Exception: java.lang.OutOfMemoryError
>        at android.graphics.BitmapFactory.nativeDecodeStream(BitmapFactory.java)
>        at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:623)
>        at android.graphics.BitmapFactory.decodeFile(BitmapFactory.java:378)
>        at android.graphics.BitmapFactory.decodeFile(BitmapFactory.java:417)
>        at com.frosquivel.magicalcamera.Functionallities.ActionPicture.onTakePhotoResult(ActionPicture.java:183)
>        at com.frosquivel.magicalcamera.Functionallities.ActionPicture.resultPhoto(ActionPicture.java:203)
>        at com.frosquivel.magicalcamera.MagicalCamera.resultPhoto(MagicalCamera.java:206)

This is other crash information:
> # Platform: android
> # Date: 2017-09-26T08:42:00Z
> # OS Version: 4.2.2
> # Device: Galaxy S2 Plus
> # RAM Free: 35.6%
> # Disk Free: 62.8%

I have this params set
> private int RESIZE_PHOTO_PIXELS_PERCENTAGE = 20;

Can you help me?

I belive you must add this function destroyPhoto:

if(bitmap!=null)
{
   bitmap.recycle();
   bitmap=null;
}

and we call this on onDestroy(), like magicalCamera.destroyPhoto();

Thanks in advances.

**Comment by alexxsanchezm** [comment]
Here I found [this](https://github.com/fabian7593/MagicalCamera/issues/16).
I don't think this a good solution but it may work for your needs.

I think the best solution should be inside the library, every time it needs to take a picture, dump the all bitmap instances and start over. For now I will try to call this feature from a separated activity or fragment. this way it will dump any reference when I leave the activity or fragment.

I tried to move the Magical camera functionality to activity or fragment, but later i think its just add more complex to the code. So I rather remove the the bitmap from the Image view and start the camera again. Always clear any current bitmap before launching the camera again. that solves the problem
